### PR TITLE
ARROW-11023: [C++][CMake] Fix gRPC build issue

### PR DESCRIPTION
**Pull request by cyb70289**
gRPC, absl, cares don't respect CMAKE_C_COMPILER and CMAKE_CXX_COMPILER
options and lead to build error if non system default compiler is used.

### CODE REVIEW

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -2366,8 +2368,6 @@ macro(build_grpc)
   set(GRPC_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/grpc_ep-install")
   set(GRPC_HOME "${GRPC_PREFIX}")
   set(GRPC_INCLUDE_DIR "${GRPC_PREFIX}/include")
-  set(GRPC_CMAKE_ARGS ${EP_COMMON_CMAKE_ARGS} "-DCMAKE_INSTALL_PREFIX=${GRPC_PREFIX}"
```
Comment (cyb70289) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:2369]: GRPC_CMAKE_ARGS is set again at line 2442

**Comment by kou** [comment]
Can we use `EP_COMMON_CMAKE_ARGS` for them?

**Comment by cyb70289** [comment]
> Can we use `EP_COMMON_CMAKE_ARGS` for them?

I think it's okay. I just see many other codes are setting CMAKE_C_COMPILER explicitly.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 2]
```diff
@@ -2449,8 +2445,6 @@ macro(build_grpc)
       -DgRPC_PROTOBUF_PROVIDER=package
       -DgRPC_SSL_PROVIDER=package
       -DgRPC_ZLIB_PROVIDER=package
-      -DCMAKE_CXX_FLAGS=${GRPC_CMAKE_CXX_FLAGS}
```
Comment (cyb70289) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:2452]: oops, should not remove this!

Comment (kou) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake:2452]: We will be able to remove this in #8881 . :-)
#8881 uses gRPC 1.33.2.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**